Add Python 3.8 & 3.9 as supported versions

kaxil opened the pull request:
Since Py 3.8 is supported, we should update the "classifiers", as it is shown on PyPI and gives wrong impressions that Py 3.8 is not supported.

This PR also adds Py 3.9 and adds them to CI

hugovk commented:
How about adding Python 3.9 as well, and testing both 3.8 and 3.9 on the CI too?

kaxil commented:
@hugovk Done

hugovk reviewed (approved):
Please could someone from this organisation enable/transfer Travis CI from .org to .com?

jsadn commented:
@jtroussard any chance you can get this merged with the impending python EOL

hugovk commented:
Which impending EOL?

Python 3.10 is due out _today_, we could also add 3.10 too :) (Would need to be `3.10-dev` on Travis CI until they add the 3.10.0 release in a few weeks' time.)

https://twitter.com/pyblogsal/status/1442890650303664131

kaxil commented:
> Which impending EOL?
> 
> Python 3.10 is due out _today_, we could also add 3.10 too :) (Would need to be `3.10-dev` on Travis CI until they add the 3.10.0 release in a few weeks' time.)
> 
> https://twitter.com/pyblogsal/status/1442890650303664131

tbh that can be done in a follow-up PR too, we shouldn't need to wait for it.

jsadn commented:
> Which impending EOL?

I was referring 3.6 in December 2021. 
3.7 is on the supported list I suppose, but I'd love to use this lib with confidence for py 3.8.

jtroussard commented:
@JonathanHuot would you be able to take a peek on these conflicts? I believe this PR should be an easy review/merge. Correct me if I am wrong but we have agreed to move away from travisCI? If so we can remove that file from this PR.

kaxil commented:
> @JonathanHuot would you be able to take a peek on these conflicts? I believe this PR should be an easy review/merge. Correct me if I am wrong but we have agreed to move away from travisCI? If so we can remove that file from this PR.

I have rebased and fixed conflicts, hopefully this time around this will be merged :)

jtroussard reviewed (approved):
lgtm
